crates/pathfinder/src/cairo/ext_py/service.rs

```diff
@@ -162,3 +178,32 @@ pub async fn start(
 
     Ok((handle, jh))
 }
+
+/// Returns if a new subprocess should be launched without wait
+fn on_joined_subprocess(
+    res: Result<Result<super::SubprocessExitInfo, anyhow::Error>, tokio::task::JoinError>,
+    processes_failed: &metrics::Counter,
+) -> bool {
```

Comment:
This is ... quite ugly but I couldn't figure out to otherwise refactor this away.

Reply:
`processes_failed` could be accessed via the `metrics::increment_counter` macro which would shorten the signature a bit. Otherwise - what else could you do :shrug: ?

Reply:
That's not the issue, returning a bool is.

Reply:
We were thinking about 2 different things, np :+1: